core/blockchain_reader.go

```diff
@@ -400,6 +400,38 @@ func (bc *BlockChain) SubscribeDirtyAccountEvent(ch chan<- []*types.DirtyStateAc
 	return bc.scope.Track(bc.dirtyAccountFeed.Subscribe(ch))
 }
 
+func (bc *BlockChain) WriteInternalTransactions(hash common.Hash, internalTxs []*types.InternalTransaction) {
+	// cache first
+	bc.internalTransactionsCache.Add(hash, internalTxs)
+
+	// check if store internal transactions is enabled or not
+	if !rawdb.ReadStoreInternalTransactionsEnabled(bc.db) {
+		return
+	}
+	rawdb.WriteInternalTransactions(bc.db, hash, internalTxs)
```

Comment:
The internal transactions are written to db and never deleted. I think we should cache them to LRU cache and store them to DB only on blockchain.Stop, load them from DB when block.NewBlockChain. The same logic can apply to dirty accounts.

Reply:
I think if we have the options of storing internal transactions within db. We should accept the increment of the storage. I agree with the logic, storing dirtyAccounts to db  when blockchain is stop